R/plotly_build.R

```diff
@@ -131,28 +131,24 @@ plotly_build.plotly <- function(p, registerFrames = TRUE) {
       class = oldClass(x)
     )
     
+    # determine trace type (if not specified, can depend on the # of data points)
+    # note that this should also determine a sensible mode, if appropriate
+    trace <- verify_type(trace)
+    # verify orientation of boxes/bars
+    trace <- verify_orientation(trace)
+    
```

Comment:
This chunk was moved here from lines 150-154, so we can be sure we know the trace type before applying `has_groups(trace)`